src/qgis_geonode/conf.py

```diff
@@ -251,5 +254,146 @@ def is_current_connection(self, connection_id: uuid.UUID):
+                    (
+                        models.GeonodeResourceType.VECTOR_LAYER
+                        in current_filters.layer_types
+                    ),
+                )
+                settings.setValue(
+                    "resource_types_raster",
+                    (
+                        models.GeonodeResourceType.RASTER_LAYER
+                        in current_filters.layer_types
+                    ),
+                )
+                settings.setValue(
+                    "resource_types_map",
+                    (models.GeonodeResourceType.MAP in current_filters.layer_types),
+                )
+            if current_filters.temporal_extent_start is not None:
+                settings.setValue(
+                    "temporal_extent_start",
+                    current_filters.temporal_extent_start.toString(QtCore.Qt.ISODate),
+                )
+            else:
+                settings.setValue("temporal_extent_start", None)
+            if current_filters.temporal_extent_end is not None:
+                settings.setValue(
+                    "temporal_extent_end",
+                    current_filters.temporal_extent_end.toString(QtCore.Qt.ISODate),
+                )
+            else:
+                settings.setValue("temporal_extent_end", None)
+
+            if current_filters.publication_date_start is not None:
+                settings.setValue(
+                    "publication_date_start",
+                    current_filters.publication_date_start.toString(QtCore.Qt.ISODate),
+                )
+            else:
+                settings.setValue("publication_date_start", None)
+            if current_filters.publication_date_end is not None:
+                settings.setValue(
+                    "publication_date_end",
+                    current_filters.publication_date_end.toString(QtCore.Qt.ISODate),
+                )
+            else:
+                settings.setValue("publication_date_end", None)
+            if current_filters.spatial_extent is not None:
+                settings.setValue(
+                    "spatial_extent_north", current_filters.spatial_extent.yMaximum()
+                )
+                settings.setValue(
+                    "spatial_extent_south", current_filters.spatial_extent.yMinimum()
+                )
+                settings.setValue(
+                    "spatial_extent_east", current_filters.spatial_extent.xMaximum()
+                )
+                settings.setValue(
+                    "spatial_extent_west", current_filters.spatial_extent.xMinimum()
+                )
+            settings.setValue("sort_by_field", current_filters.ordering_field.value)
+            settings.setValue("reverse_sort_order", current_filters.reverse_ordering)
+
+    def get_current_search_filters(self) -> models.GeonodeApiSearchParameters:
+        with qgis_settings(
+            f"{self.BASE_GROUP_NAME}/{self.SEARCH_GROUP}/current_search_filters"
```

Comment:
According with the previous comment, please do not use such a deep hierarchy here
```suggestion
            f"{self.BASE_GROUP_NAME}/current_search_filters"
```